src/pyre_refining.cc

```diff
@@ -0,0 +1,199 @@
+#include "pyre.h"
+#include "process.h"
+#include "pyre_refining.h"
+#include "boost/bind.hpp"
+#include <boost/math/tools/roots.hpp>
+
+using cyclus::Material;
+using cyclus::Composition;
+using cyclus::toolkit::ResBuf;
+using cyclus::toolkit::MatVec;
+using cyclus::KeyError;
+using cyclus::ValueError;
+using cyclus::Request;
+using cyclus::CompMap;
+
+using boost::math::tools::bisect;
+
+namespace recycle {
+
+Refine::Refine() {
+  temp(0);
+  pressure(0);
+  rotation(0);
+  b_size(0);
+  Rtime(0);
+}
+
+Refine::Refine(double new_temp = 900, 
+               double new_press = 760, 
+               double new_rotation = 0, 
+               double new_batch_size = 20,
+               double new_rtime = 1
+            ) 
+            {
+  SetCoeff();
+  temp(new_temp);
+  pressure(new_press);
+  rotation(new_rotation);
+  b_size(new_batch_size);
+  Rtime(new_rtime);
+}
+
+void Refine::SetCoeff() {
+  t0_ = 4.7369E-9;
+  t1_ = -1.08337E-5;
+  t2_ = 0.008069;
+  t3_ = -0.9726;
+  p0_ = -7.17631E-10;
+  p1_ = 4.04545E-07;
+  p2_ = -8.06336E-05;
+  p3_ = 1.002;
+  a0_ = 0.032;
+  a1_ = 0.72;
+  a2_ = 0.0338396;
+  a3_ = 0.83667;
+}
+
+struct Refine::TerminationCondition {
+    bool operator() (double min, double max)  {
+        return abs(min - max) <= 0.000001;
+    }
+};
+
+void Refine::DivertMat(std::string type, std::pair<std::string, std::string> location,
+  double siphon) {
+    if (type == "operator") {
+      OpDivertMat(location, siphon);
+    } 
+}
+
+void Refine::OpDivertMat(std::pair<std::string, std::string> location, double siphon) {
+  std::string param = location.second;
+  double newVal;
+
+  double paramVal = subcomponents[param].back();
+  double lower_bound = paramVal-(paramVal*0.1);
+  double upper_bound = paramVal+(paramVal*0.1);
+  if (param == "temp"){
+    double newEff = Thermal() * (1+siphon);
+    ThermalFunc = boost::bind(&recycle::Refine::Thermal,this,_1,newEff);
+    newVal = Bisector(ThermalFunc, lower_bound, upper_bound);
+  } else if (param == "pressure") {
+  } else if (param == "rotation") {
+  } else if (param == "batch") {
+  } else {
+  }
+  subcomponents[param].push_back(newVal);
+}
+
+double Refine::Bisector(boost::function<double(double)> SolveFunc, double lower, double upper) {
+  std::pair<double, double> result = bisect(ThermalFunc, 
+      temp()-100.0, temp()+100.0, TerminationCondition());
+  return (result.first + result.second) / 2;
+}
+
+double Refine::Efficiency() {
+  return Thermal()*PressureEff()*Agitation();
+}
+
+double Refine::Thermal() {
+  return t0()*pow(temp(),3) + t1()*pow(temp(),2) + t2()*temp() + t3();
+}
+
+double Refine::Thermal(double tmp,
+                       double newEff = 0
+) {
+  return t0()*pow(tmp,3) + t1()*pow(tmp,2) + t2()*tmp + t3() - newEff;
+}
+
+double Refine::PressureEff() {
+  return p0()*pow(pressure(), 3) + p1()*pow(pressure(), 2) + p2()*pressure() + p3();
```

Comment:
why not calling the next method `PressureEff(double, double)` passing the member variable (so you have only one implementation of the function -> so only one to maintain :)